### DISCUSSION TITLE
One-Click Droplet via DigitalOcean results in 502 Error

### DISCUSSION THREAD

**Discussion by Snokown** [directus/directus#16382] [Archive]
### Describe the Bug

Creating a new droplet using the one-click install in the DigitalOcean marketplace completes without errors but access to the admin app is unreachable and shows a 502 error.

### To Reproduce

Create a new droplet and use the directus one-click option in the marketplace.

### Errors Shown

502 Server error

### What version of Directus are you using?

9/latest

### What version of Node.js are you using?

14

### What database are you using?

Postgres 12

### What browser are you using?

Chrome

### How are you deploying Directus?

DigitalOcean One-Click Install

**Comment by azrikahar** [comment]
Tried it out and can confirm it results in 502 error. Findings:

Once I ssh into the droplet, I ran `netstat -tulpn` (after installing `net-tools`) and based on the result, it seems like directus is not running (there should be a process listening to port `8055`):

```
root@directus9onubuntu1804-s-1vcpu-2gb-amd-sgp1-01:/var/directus# netstat -tulpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      633/systemd-resolve
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      817/sshd: /usr/sbin
tcp        0      0 0.0.0.0:8055            0.0.0.0:*               LISTEN      9487/node
tcp        0      0 127.0.0.1:5432          0.0.0.0:*               LISTEN      866/postgres
tcp        0      0 127.0.0.1:2019          0.0.0.0:*               LISTEN      3493/caddy
tcp6       0      0 :::22                   :::*                    LISTEN      817/sshd: /usr/sbin
tcp6       0      0 :::80                   :::*                    LISTEN      3493/caddy
udp        0      0 127.0.0.53:53           0.0.0.0:*                           633/systemd-resolve
```

I then `cd` into `/var/directus` to attempt to start it with `npm start` (the start script in package.json is `directus start`), which then shows the following error:

```
root@directus9onubuntu1804-s-1vcpu-2gb-amd-sgp1-01:/var/directus# npm start

> project@1.0.0 start /var/directus
> directus start

internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'node:path'
Require stack:
- /var/directus/node_modules/rollup/dist/shared/rollup.js
- /var/directus/node_modules/rollup/dist/rollup.js
- /var/directus/node_modules/directus/dist/extensions.js
- /var/directus/node_modules/directus/dist/services/graphql/index.js
- /var/directus/node_modules/directus/dist/services/index.js
- /var/directus/node_modules/directus/dist/middleware/respond.js
- /var/directus/node_modules/directus/dist/controllers/activity.js
- /var/directus/node_modules/directus/dist/app.js
- /var/directus/node_modules/directus/dist/server.js
- /var/directus/node_modules/directus/dist/cli/index.js
- /var/directus/node_modules/directus/dist/cli/run.js
- /var/directus/node_modules/directus/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/var/directus/node_modules/rollup/dist/shared/rollup.js:12:19)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/var/directus/node_modules/rollup/dist/shared/rollup.js',
    '/var/directus/node_modules/rollup/dist/rollup.js',
    '/var/directus/node_modules/directus/dist/extensions.js',
    '/var/directus/node_modules/directus/dist/services/graphql/index.js',
    '/var/directus/node_modules/directus/dist/services/index.js',
    '/var/directus/node_modules/directus/dist/middleware/respond.js',
    '/var/directus/node_modules/directus/dist/controllers/activity.js',
    '/var/directus/node_modules/directus/dist/app.js',
    '/var/directus/node_modules/directus/dist/server.js',
    '/var/directus/node_modules/directus/dist/cli/index.js',
    '/var/directus/node_modules/directus/dist/cli/run.js',
    '/var/directus/node_modules/directus/cli.js'
  ]
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! project@1.0.0 start: `directus start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the project@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-11-07T04_55_59_263Z-debug.log
```

Note that the `node:path` import error stems from `rollup`. This is because `rollup` was upgraded from `^2.75.6` → `3.2.3` in #16052, and rollup had a breaking change in [`3.0.0`](https://github.com/rollup/rollup/releases/tag/v3.0.0) specifically due to the node prefixed import:

![](https://user-images.githubusercontent.com/42867097/200231322-2117416e-c727-4ee5-b132-5f5e3a7ffcd6.png)

which meant it now requires a minimum of Node `14.18.0` to run. However the [Directus 1-Click App](https://marketplace.digitalocean.com/apps/directus) uses Node `14.15.4` (the web page only shows `14`, but verified in the droplet via terminal).

I ran the following commands:

```
sudo apt update
sudo apt install nodejs
```

and it seems to update Nodejs up to `14.21.1`, and the `node:path` import error doesn't seem to occur anymore.

However I'm not 100% are there any other issues, as `npx directus start` doesn't seem to work properly. I believe there's some form of error, but it is not shown due to #16299, so there may need to be additional investigation if bumping the Node version of the 1-click install doesn't 100% resolve this issue.

**Reply by Nvironmental** [reply]
Hey @azrikahar,
Hope you are doing well. 


I was facing the same issue and after hours of failing and crying in the corner, I found a very simple solution :) 

After running these,
```
sudo apt update
sudo apt install nodejs
```

We have to run 
```
npx directus bootstrap
```
This will re-initialise the database (do note the Email and Password that it will generate at the end of bootstrapping) 

Then, we can simply run 

```
npx directus start
```

And voila, the admin panel is up and running :) 

Hope this helps and gets documented by the Directus Team

**Comment by lucameusburger** [comment]
Are there any updates regarding the one-click-droplet. Also you can't find it on the marketplace but it is still there on https://docs.digitalocean.com/products/marketplace/catalog/directus/ .

**Reply by rijkvanzanten** [reply]
I'd strongly recommend using the Docker Container on App Platform instead of the one-click. The one-click isn't actively maintained 👍🏻

**Reply by lucameusburger** [reply]
Thank you so much! What a easy setup with docker.. I was tinkering around with a app-platform-deploy and the one-click-deploy for too long!

**Reply by Snokown** [reply]
Thanks @rijkvanzanten - I went with a docker + caddy solution and it worked out great. Going to be my go-to solution from now on 👍🏻

**Reply by rijkvanzanten** [reply]
Docker is such a better way to deploy things nowadays that I'm seriously considering dropping the npm way of installing directus from the docs. It's needlessly confusing if you can just "docker run" and be done! :o
